handler/src/main/java/io/netty/handler/ssl/SslHandler.java

```diff
@@ -574,11 +575,17 @@ public SSLEngine engine() {
      */
     public String applicationProtocol() {
         SSLSession sess = engine().getSession();
-        if (!(sess instanceof ApplicationProtocolAccessor)) {
-            return null;
+        if (sess instanceof ApplicationProtocolAccessor) {
+            return ((ApplicationProtocolAccessor) sess).getApplicationProtocol();
+        }
+        if (Java9SslUtils.supportsAlpn(engine())) {
```

Comment:
@normanmaurer or was the issue with classloading only if we invoke a method which contains a class which isn't on the class path?